model/group.cc

```diff
@@ -411,7 +409,7 @@ namespace minsky
       return IORegion::output;
     else if (-w+left*edgeScale()>dx)
       return IORegion::input;
-    else if (-h+10*z>dy || h-10*z<dy)     
+    else if ((fabs(-h-10*z)>fabs(dy) && dy<0) || (h+10*z>dy && dy>0))     
```

Comment:
I'm suspicious of this line. The two clauses aren't symmetric.

Reply:
The only thing I could fix. Have been working hard on feature 94 and made some progress. I hope to finish it by tomorrow but am not sure I'll get to ticket 1173 before Thursday, but I'll try :)